pkg/util/ftp/auth.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ftp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+)
+
+const (
+	secretKeyWithAPIKey      = "apikey"
+	secretKeyWithBearerToken = "token"
+	secretKeyWithUsername    = "username"
+	secretKeyWithPassword    = "password"
+)
+
+// FileServerAuthType defines the authentication type to access a file server.
+type FileServerAuthType string
+
+const (
+	APIKey              FileServerAuthType = "APIKey"
+	BearerToken         FileServerAuthType = "BearerToken"
+	BasicAuthentication FileServerAuthType = "BasicAuthentication"
+)
+
+type FileServerAuthConfiguration struct {
```

Comment:
The auth parser part is not necessary bound with a file server, maybe remove `FileServer` from the var name,
I suppose we can move this to a separate package? maybe 'auth' under the 'util' folder? @antoninbas @tnqn

Reply:
updated